### DISCUSSION TITLE
Beginner to interpret the output of a 30-binary, 3-objective problem

### DISCUSSION THREAD

**Discussion by dinohsu1019** [anyoptimization/pymoo#514] [Q&A] [answered]
Attached is a problem with 30 binary variables and 3 objectivs:

https://tinyurl.com/ymdz7yr9

A few questions about the output:

Do we look at number of solutions in the Pareto Front only? Do we ignore Epsilon and indicator columns completely? (do not see any usage of it)

The number of solutions in the Pareto Front increases initially and decreases later, is this normal?

Is the number of solutions in the Pareto Front always more than zero?

During minimize() execution, how do we know if we have good solutions from output or is it feasible to?

Is it possible to continue the minimize() after it finishes? If it is with long execution time, we probably want to make sure we have got good solutions before it stops.

Thanks in advance.

**Comment by blankjul** [comment, marked as answer]
Yes you can look only into the Pareto front. The size will be at least one for unconstrained problems.

You can use a `Callback` (https://pymoo.org/interface/callback.html) to track how many solutions are feasible.

You can not `minimize` but start it again with a biased population (you can use here the result returned by your first `minimize` call)
https://pymoo.org/customization/initialization.html

**Reply by dinohsu1019** [reply]
Just to summarize a few points that I've learned on top of the above answer and the documentation:
1. Pareto size is at least one if there's no constraint; the reason the Parento sizes increase and decrease over time, I think, is the nature of discrete problems, in other words, the design (solution) space is not "smooth" and can result in sudden changes in the objective space.
2. Sorry I have caused some confusion in my question with "feasibe", what I mean is whether I can know solutions in this generation is good enough, and I think we should look at the objective values (pop.get("F").
3. For feasibility of solutions, I try not to introduce constraints at this momnet, but not sure will be able to keep it that way.
4. Callback, ask/tell, has_next/next are some controls over each generation.
5. Biased initialization with array and population options are a function for continuation of the search.
6. In my specific application, an evluation is as slow as 14 seconds, and now I have adopted objective space termination with tolerance=0.001 and window=10, which can terminate without haveing to reach the maximum generations.
7. For diversity of at least within the same population, I think eliminate_duplicates = True makes sense.

**Reply by blankjul** [reply]
1. It increases because more non-dominated solutions are found when finally converging. Look up convergence and diversity in MOO for more information.

Yes to everything else! If an evaluation is 14 seconds you can also check if you can parallelize in thread or processes if this is not already happening.

**Reply by dinohsu1019** [reply]
I have tried Starmap parallelization and found it easy to use, the only issue is the core function of evaluation is a 3rd party library (for backtesting), and it is not working with parallelization, I have submitted a request for the parallelization upgrade. In the meantime, In my 12-core Windows + Intel notebook, it doesn't seem to go faster with n_threads=4 and 12, but maybe it's that non-parallelization-aware function that causes resource contention and slows it down. I will study convergence and diversity of MOO and see I can know more. Thanks again.
